src/+otp/+pendulumdae/PendulumDAEProblem.m

```diff
@@ -0,0 +1,87 @@
+classdef PendulumDAEProblem < otp.Problem
+    %CONSTRAINEDPENDULUM PROBLEM This is a Hessenberg Index-2 DAE problem posed in terms of three constraints
+    %
+    
+    properties (SetAccess=protected)
+        RHSDifferential
+        RHSAlgebraic
+    end
+
+    properties (Dependent)
+      Y0Differential
+      Z0Algebraic
+    end
+
+    methods
+        function obj = PendulumDAEProblem(timeSpan, y0, parameters)
+            obj@otp.Problem('Constrained Pendulum', 7, timeSpan, y0, parameters);
+        end
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+            m = obj.Parameters.Mass;
+            l = obj.Parameters.Length;
+            g = obj.Parameters.Gravity;
+
+            % get initial energy
+            initialconstraints = otp.pendulumdae.invariants([], obj.Y0Differential, g, m, l, 0);
+            E0 = initialconstraints(3);
+
+            % The right hand size in terms of x, y, x', y', and three control parameters z
+            obj.RHS = otp.RHS(@(t, y) otp.pendulumdae.f(t, y, g, m, l, E0), ...
+                'Mass', otp.pendulumdae.mass([], [], g, m, l, E0), ...
+                'JacobianVectorProduct', ...
+                @(t, y, v) otp.pendulumdae.jacobianVectorProduct(t, y, g, m, l, E0, v), ...
+                'Vectorized', 'on');
+
+            % Generate the constituent RHS for the differential part
+            obj.RHSDifferential = otp.RHS(@(t, y) otp.pendulumdae.fDifferential(t, y, g, m, l, E0), ...
+                'JacobianVectorProduct', ...
+                @(t, y, v) otp.pendulumdae.jacobianDifferentialVectorProduct(t, y, g, m, l, E0, v), ...
```

Comment:
`jacobianVectorProductDifferential` is the naming convention https://github.com/ComputationalScienceLaboratory/ODE-Test-Problems/blob/master/docs/CONTRIBUTING.md#functions-for-rhs